transaction/types/Cargo.toml

```
@@ -5,12 +5,17 @@ authors = ["MobileCoin"]
 edition = "2021"
 readme = "README.md"
 
+[features]
+serde = ["dep:serde"]
+alloc = ["serde/alloc"]
```

This unconditionally includes `serde`. We can make it only turn on `serde/alloc` when `serde` is activated with `?/` (since Rust 1.60)
```suggestion
alloc = ["serde?/alloc"]
```